client/src/App.js

```diff
@@ -1,18 +1,23 @@
-import React from "react";
+import React, { useState } from "react";
 import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
 import Home from "./pages/Home";
 import Welcome from "./pages/Welcome";
 import List from "./pages/List";
 import GlobalStyles from "./GlobalStyles";
+import Compare from "./pages/Compare";
 
 function App() {
+  const [listData, setListData] = useState("");
   return (
     <>
       <GlobalStyles />
       <Router>
         <Switch>
+          <Route path="/home/:id/compare">
+            <Compare listData={listData} />
+          </Route>
           <Route path="/home/:id">
-            <List />
+            <List setListData={(listData) => setListData(listData)} />
```

Comment:
There is an issue with the naming here.
In HTML (and React) event listeners starts with `on` like `onChange` or `onClick`. 
In this case, I would suggest this:
```suggestion
            <List onListDataChange={(listData) => setListData(listData)} />
```
This is more common. 

What is `listData` anyway? 😃

Reply:
That's a good point here. I will change it as suggested. listData is the grocery list you are editing as I renamed it now (groceryList)

Reply:
listData will have my grocery List as its content . I renamed it to the latter. Thank you for your tips